src/main/java/com/github/webicitybrowser/spec/fetch/imp/FetchEngineImp.java

```diff
@@ -41,8 +43,17 @@ private FetchResponse httpNetworkFetch(FetchParams params) {
 
 	private void fetchResponseHandover(FetchParams params, FetchResponse response) {
 		if (params.consumeBodyAction() != null) {
-			// TODO: Queue a response
+			if(response.body() == null) {
+				// TODO: Queue a response
+				queueAFetchTask(params, response);
+			} else {
+				params.consumeBodyAction().execute(response, false, new byte[]{});
+			}
 		}
 	}
-	
+
+	private void queueAFetchTask(FetchParams params, FetchResponse response) {
+		params.taskDestination().enqueue(params.consumeBodyAction());
```

Comment:
The queue a fetch task procedure is also used for things other than the consume body action
You should probably accept a Runnable instead of a FetchParams
For example, `queueAFetchTask(() -> params.consumeBodyAction(), response)`